### PR TITLE
add the possibility to use a different field as the Content Provider _id

### PR DESCRIPTION
**Pull request by robUx4**
Fixes https://github.com/BoD/android-contentprovider-generator/issues/56

### CODE REVIEW

File: src/main/java/org/jraf/androidcontentprovidergenerator/Main.java [hunk 1]
```diff
@@ -128,7 +124,7 @@ public boolean accept(File pathname) {
                 if (fieldDocumentation.isEmpty()) fieldDocumentation = null;
                 String type = fieldJson.getString(Field.Json.TYPE);
                 boolean isIndex = fieldJson.optBoolean(Field.Json.INDEX, false);
-                boolean isNullable = fieldJson.optBoolean(Field.Json.NULLABLE, true);
+                boolean isNullable = !isIndex && fieldJson.optBoolean(Field.Json.NULLABLE, true);
```
Comment (BoD) [inline src/main/java/org/jraf/androidcontentprovidergenerator/Main.java]: Why?

File: src/main/java/org/jraf/androidcontentprovidergenerator/Main.java [hunk 2]
```diff
@@ -157,11 +153,39 @@ public boolean accept(File pathname) {
...
+
+                if (idField == null) {
+                    throw new IllegalArgumentException("Invalid idField '" + idFieldName + "' value in " + entityFile.getCanonicalPath() + ".");
+                } else if (idField.getType() != Field.Type.INTEGER && idField.getType() != Field.Type.LONG &&
+                        idField.getType() != Field.Type.DATE && idField.getType() != Field.Type.ENUM){
+                    throw new IllegalArgumentException("Invalid idField type " + idField.getType() + " in " + entityFile.getCanonicalPath() + "."
+                            + " It can only be Integer, Long, Date, Enum.");
+                } else if (idField.getIsNullable()) {
+                    throw new IllegalArgumentException("Invalid idField '" + idFieldName + "' is nullable in " + entityFile.getCanonicalPath() + ".");
+                } else if (!idField.getIsIndex() && !idField.getIsId()) {
```
Comment (BoD) [inline src/main/java/org/jraf/androidcontentprovidergenerator/Main.java]: I don't understand the check for getIsId here.

**Comment by BoD** [comment]
Thank you for this long and tedious contribution :p
Please make sure it still works after my numerous changes and "fixes".

File: src/main/java/org/jraf/androidcontentprovidergenerator/Main.java (hunk 1 above)
Comment (robUx4) [inline src/main/java/org/jraf/androidcontentprovidergenerator/Main.java]: Having an index an a null value sound weird to me (ie a non existant value). I don't know if SQlite allows it, but why not.

File: src/main/java/org/jraf/androidcontentprovidergenerator/Main.java (hunk 2 above)
Comment (robUx4) [inline src/main/java/org/jraf/androidcontentprovidergenerator/Main.java]: In case the _id field is not automatically created.

**Pull request merged**